src/yaml/yamlUtils.ts

```diff
@@ -308,3 +311,148 @@ export function convertJsonSnippetToYaml(snippetText: string) {
         })
         .join('\n')
 }
+
+function getNumberOfLeftSpaces(text: string) {
+    return text.length - text.trimLeft().length
+}
+
+function getBackwardOffsetData(text: string, offset: number, initialNumberOfSpaces: number) {
+    let isDirectChildOfStates = false
+    let isGrandChildOfStates = false
+    let isWithinCatchRetryState = false;
+    let hasCatchPropSibling = false;
+    let hasRetryPropSibling = false;
+    let beginLineOffset = offset;
+    let levelsDown = 0;
+
+    for (let i = offset; i >= 0; i--) {
+        if (text[i] === '\n') {
+            const lineText = text.slice(i + 1, beginLineOffset)
+            const numberOfPrecedingSpaces = getNumberOfLeftSpaces(lineText)
+            const trimmedLine = lineText.trim()
+            beginLineOffset = i
+
+            // Ignore empty lines
+            if (trimmedLine.length === 0) {
+                continue
+            }
+
+            // If number of spaces lower than that of the cursor
+            // it is a parent property or a sibling of parent property
+            if (numberOfPrecedingSpaces < initialNumberOfSpaces) {
+                if (trimmedLine.startsWith('States:')) {
+                    isDirectChildOfStates = levelsDown === 0;
+                    isGrandChildOfStates = levelsDown === 1;
+                    break
+                } else if (levelsDown === 0) {
```

Comment:
Can you provide some additional comments about the behaviour taken in the else cases?